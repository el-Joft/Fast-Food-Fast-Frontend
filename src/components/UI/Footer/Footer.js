import React from 'react';
import { Link } from 'react-router-dom';
import './Footer.scss';
import { CURRENT_YEAR } from '../../../helpers/helpers';

const Footer = () => (
  <footer>
    <div className="footer footer-home">
      <div className="contact-form">
        <div className="contact-title">Contact Us</div>
        <form>
          <ul className="form-group">
            <li>
              <label>
Full Name :
                {' '}
                <span className="required">*</span>
              </label>
              <input type="text" name="text" className="form-control field-long" required />
            </li>
            <li>
              <label>
Email
                {' '}
                <span className="required">*</span>
              </label>
              <input type="email" name="email" className="form-control field-long" required />
            </li>
            <li>
              <label>
Your Message
                {' '}
                <span className="required">*</span>
              </label>
              <textarea name="message" id="message" rows="3" required="required" className="field-long field-textarea" />
            </li>
            <li>
              <button type="submit" className="button"><span>Send</span></button>
            </li>
          </ul>
        </form>
      </div>
      <div className="footer-navigation"><Link to="/admin">Admin?... click here</Link></div>
      <h5 className="footer-text">
&copy;
        {' '}
        <span id="year">{CURRENT_YEAR}</span>
        {' '}
Andela bootcamp project
        {' '}
      </h5>
    </div>
  </footer>
);

export default Footer;
