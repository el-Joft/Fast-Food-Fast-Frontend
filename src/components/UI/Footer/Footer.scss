/* FOOTER */

.footer{
  margin-top: 10px;
  padding: 10px;
  height: 150px;
  background: #42ce9f;
 text-align: center;
}
.footer-home{
  height: 500px!important;
}
.footer-navigation a{
  color: #fff;
}
.contact-form{
  width: 100%;
}
.contact-title{
  font-size: 1.3em;
  margin-bottom: 12px;
  color: #fff;
}
.button{
  border: 2px solid #42ce9f;
  color:#42ce9f;
  background: #fff;
  padding: 10px;
  font-size: 1.0em;
  border-radius: 30px;
}
.footer-social{
  display:flex;
 
}
.footer-social li{
  margin-right:10px;
}
.footer-social img{
  width: 40px;
  height: 40px;

}
.footer-text{
  text-align: center;
  font-size: 1.2em;
  color: #fff;
  font-family: Handlee;
  margin-top: 12px;
  padding: 10px;
}


/* FORM */

.form-control {
  width: 30%;
  padding: 10px;
  margin-bottom: 10px;
  font-size: 1rem;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  border: 1px solid #42ce9f;
  border-radius: .25rem;
  transition: border-color .15s ease-in-out,box-shadow .15s ease-in-out;
 
  
}

.form-group {
  margin:10px auto;
  max-width: 400px;
  padding: 20px 12px 10px 20px;
  font: 13px "Lucida Sans Unicode", "Lucida Grande", sans-serif;
}
.form-group li {
  padding: 0;
  display: block;
  list-style: none;
  margin: 10px 0 0 0;
}
.form-group label{
  margin:0 0 3px 0;
  padding:0px;
  display:block;
  font-weight: bold;
  font-size: 1.7;
}

.select {
  border: 1px solid #42ce9f;
}
option {
  color: #42ce9f;
}

.form-group .field-long{
  width: 100%;
}

.form-group input[type=submit], .form-group input[type=button]{
  background: #42ce9f;
  padding: 8px 15px 8px 15px;
  border: none;
  color: #fff;
}
.form-group .field-divided{
  width: 49%;
}
.form-group .required{
  color:red;
}
/* this aspect was learnt from the MDN */
/* https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Styling_HTML_forms */
/* MORE ON FORMS */
.form-group input[type=text]:focus, 
.form-group input[type=date]:focus,
.form-group input[type=datetime]:focus,
.form-group input[type=number]:focus,
.form-group input[type=search]:focus,
.form-group input[type=time]:focus,
.form-group input[type=url]:focus,
.form-group input[type=email]:focus,
.form-group input[type=password]:focus,
.form-group textarea:focus, 
.form-group select:focus{
  -moz-box-shadow: 0 0 8px #88D5E9;
  -webkit-box-shadow: 0 0 8px #88D5E9;
  box-shadow: 0 0 8px #88D5E9;
  border: 1px solid #88D5E9;
}

/* end of the aspect */



.text-form{
  text-align: center;
  padding: 10px;

}
//* DELETE BUTTON */
.btn-danger{
  background: #ff0a10;
  color: #fff;
  border: 1px solid #ff0a10;
  border-radius: 15%;
  margin: 5px;
}

.btn-danger a{
  color: #fff;
}

.btn-normal{
  background: #42ce9f;
  color: #fff;
  border: 1px solid #42ce9f;
  border-radius: 15%;
  margin: 5px;
}


.btn-normal a{
  color: #fff;
}
.complete-label{
  font-size: 12px;
  display: block;
}



/* FOOTER */

.footer{
    margin-top: 10px;
    padding: 10px;
    height: 150px;
    background: #42ce9f;
   text-align: center;
    
    
}
.footer-home{
    height: 500px!important;
    
}
.footer-navigation a{
    color: #fff;
}
.contact-form{
    width: 100%;
}
.contact-title{
    font-size: 1.3em;
    margin-bottom: 12px;
    color: #fff;
}
.button{
    border: 2px solid #42ce9f;
    color:#42ce9f;
    background: #fff;
    padding: 10px;
    font-size: 1.0em;
    border-radius: 30px;
}
.footer-social{
    display:flex;
   
}
.footer-social li{
    margin-right:10px;
}
.footer-social img{
    width: 40px;
    height: 40px;

}
.footer-text{
    text-align: center;
    font-size: 1.2em;
    color: #fff;
    font-family: Handlee;
    margin-top: 12px;
    
    padding: 10px;
}


/* FORM */

.form-control {
    width: 30%;
    padding: 10px;
    margin-bottom: 10px;
    font-size: 1rem;
    line-height: 1.5;
    color: #495057;
    background-color: #fff;
    border: 1px solid #42ce9f;
    border-radius: .25rem;
    transition: border-color .15s ease-in-out,box-shadow .15s ease-in-out;
   
    
}

.form-group {
    margin:10px auto;
    max-width: 400px;
    padding: 20px 12px 10px 20px;
    font: 13px "Lucida Sans Unicode", "Lucida Grande", sans-serif;
}
.form-group li {
    padding: 0;
    display: block;
    list-style: none;
    margin: 10px 0 0 0;
}
.form-group label{
    margin:0 0 3px 0;
    padding:0px;
    display:block;
    font-weight: bold;
    font-size: 1.7;
}

.select {
    border: 1px solid #42ce9f;
}
option {
    color: #42ce9f;
}

.form-group .field-long{
    width: 100%;
}

.form-group input[type=submit], .form-group input[type=button]{
    background: #42ce9f;
    padding: 8px 15px 8px 15px;
    border: none;
    color: #fff;
}
.form-group .field-divided{
    width: 49%;
}
.form-group .required{
    color:red;
}
/* this aspect was learnt from the MDN */
/* https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Styling_HTML_forms */
/* MORE ON FORMS */
.form-group input[type=text]:focus, 
.form-group input[type=date]:focus,
.form-group input[type=datetime]:focus,
.form-group input[type=number]:focus,
.form-group input[type=search]:focus,
.form-group input[type=time]:focus,
.form-group input[type=url]:focus,
.form-group input[type=email]:focus,
.form-group input[type=password]:focus,
.form-group textarea:focus, 
.form-group select:focus{
    -moz-box-shadow: 0 0 8px #88D5E9;
    -webkit-box-shadow: 0 0 8px #88D5E9;
    box-shadow: 0 0 8px #88D5E9;
    border: 1px solid #88D5E9;
}

/* end of the aspect */

.text-form{
    text-align: center;
    padding: 10px;
  
}


/* DELETE BUTTON */
.btn-danger{
  background: #ff0a10;
  color: #fff;
  border: 1px solid #ff0a10;
  border-radius: 15%;
  margin: 5px;
}

.btn-danger a{
  color: #fff;
}

.btn-normal{
  background: #42ce9f;
  color: #fff;
  border: 1px solid #42ce9f;
  border-radius: 15%;
  margin: 5px;
}


.btn-normal a{
  color: #fff;
}
.complete-label{
  font-size: 12px;
  display: block;
}
