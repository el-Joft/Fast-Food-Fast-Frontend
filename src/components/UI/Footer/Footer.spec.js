/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { Footer } from './Footer';

describe('rendering', () => {
  let wrapper;
  it('should render the component', () => {
    wrapper = shallow(<Footer />);
    expect(wrapper.length).toBe(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<Footer />);
    const form = wrapper.find('form');
    expect(form.length).toBe(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});
