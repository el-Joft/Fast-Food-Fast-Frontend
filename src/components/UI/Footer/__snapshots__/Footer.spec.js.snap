// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
<footer>
  <div
    className="footer footer-home"
  >
    <div
      className="contact-form"
    >
      <div
        className="contact-title"
      >
        Contact Us
      </div>
      <form>
        <ul
          className="form-group"
        >
          <li>
            <label>
              Full Name :
              <span
                className="required"
              >
                *
              </span>
            </label>
            <input
              className="form-control field-long"
              name="text"
              required={true}
              type="text"
            />
          </li>
          <li>
            <label>
              Email
              <span
                className="required"
              >
                *
              </span>
            </label>
            <input
              className="form-control field-long"
              name="email"
              required={true}
              type="email"
            />
          </li>
          <li>
            <label>
              Your Message
              <span
                className="required"
              >
                *
              </span>
            </label>
            <textarea
              className="field-long field-textarea"
              id="message"
              name="message"
              required="required"
              rows="3"
            />
          </li>
          <li>
            <button
              className="button"
              type="submit"
            >
              <span>
                Send
              </span>
            </button>
          </li>
        </ul>
      </form>
    </div>
    <div
      className="footer-navigation"
    >
      <Link
        replace={false}
        to="/admin"
      >
        Admin?... click here
      </Link>
    </div>
    <h5
      className="footer-text"
    >
      Â©
      <span
        id="year"
      >
        2019
      </span>
       
      Andela bootcamp project
       
    </h5>
  </div>
</footer>
`;
